setup.py

```diff
@@ -43,26 +43,37 @@ def unique_flatten_dict(d):
     'jupyter': ['ipython'],
 }
 
+base_extras_data = {
+    'names-dataset': ['names-dataset'],
+    'Faker': ['Faker'],
+    'random-address': ['random-address'],
+    'phone-gen': ['phone-gen'],
```

Comment:
* roll into one: `'data-gen': ['names-dataset', 'faker', 'random-address']`
* also, try testing that `pip install x y z` for each of these names works


maybe worth getting into gha ci as well as a (small) extra test job? `pip install .data-gen`, something like that